src/components/run-button/run-button-hooks.js

```diff
@@ -0,0 +1,175 @@
+import { UPDATE_TYPE_HEADER } from 'components/study-container';
+import { RunningStatus } from 'components/utils/running-status';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { setRunButtonStatus } from 'redux/actions';
+
+function isWorthUpdate(
+    studyUpdatedForce,
+    fetcher,
+    lastUpdateRef,
+    nodeUuidRef,
+    nodeUuid,
+    invalidations
+) {
+    const headers = studyUpdatedForce?.eventData?.headers;
+    const updateType = headers?.[UPDATE_TYPE_HEADER];
+    const node = headers?.['node'];
+    const nodes = headers?.['nodes'];
+    if (nodeUuidRef.current !== nodeUuid) {
+        return true;
+    }
+    if (fetcher && lastUpdateRef.current?.fetcher !== fetcher) {
+        return true;
+    }
+    if (
+        studyUpdatedForce &&
+        lastUpdateRef.current?.studyUpdatedForce === studyUpdatedForce
+    ) {
+        return false;
+    }
+    if (!updateType) {
+        return false;
+    }
+    if (invalidations.indexOf(updateType) <= -1) {
+        return false;
+    }
+    if (node === undefined && nodes === undefined) {
+        return true;
+    }
+    if (node === nodeUuid || nodes?.indexOf(nodeUuid) !== -1) {
+        return true;
+    }
+
+    return false;
+}
+
+// this hook loads <runButtonType> state into redux, then keeps it updated according to notifications
+export function useRunButtonStatus(
+    studyUuid,
+    nodeUuid,
+    fetcher,
+    invalidations,
+    resultConversion,
+    runButtonType
+) {
+    const nodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+    const dispatch = useDispatch();
+
+    const update = useCallback(() => {
+        nodeUuidRef.current = nodeUuid;
+        fetcher(studyUuid, nodeUuid)
```

Comment:
A small comment to explain what is this `fetcher` method would be nice (we can't guess without checking the calls of this hook)
Even better, a full description of the hook, like
```
/**
 * 
 * @param studyUuid
 * @param nodeUuid
 * @param fetcher
 * @param invalidations
 * @param resultConversion
 * @param runButtonType
 */
```